packages/core/src/Collapse.ts

```diff
@@ -56,16 +62,21 @@ export class Collapse {
     const collapseElement = this.options.getCollapseElement()
     if (this.collapseElement !== collapseElement) {
       this.collapseElement = collapseElement
+      this.setInitialDimensions()
       if (!this.isExpanded) {
-        this.setStyles(this.getCollapsedStyles())
+        this.setStyles({
+          ...this.getCollapsedStyles(),
+          // When expand = false and collapsing width we preserve initial height
+          ...(this.options.isHorizontal ? {height: `${this.initialHeight}px`} : {}),
```

Comment:
I think this might be an issue with an element that's made the full height of the window. An example would be a drawer or a sidebar. Upon resizing, hardcoding the height would cause the sidebar to not stay adhered to the window edge.

We might need to use a resizeobserver in order to update the height

Reply:
Or maybe it should be an option to persist the height when the width is animated and collapsed? Not sure which is better

Reply:
There is two places where we persist the height:
- In the init method when the default expand is equal to false.
- In the close method when the width is animated and collapsed.

If we are going to use **resizeobserver** (observe height) **on the element** it will interfere with collapase when it is changing the height for animated width.

I think the second option you mentioned is already handled in close method and for me i go for this option, if you have any suggestions to improve it let me know please ;)

Reply:
I pulled up the storybook, and it looks like the fixed height is always applied (it would need to be removed in the transitionend handler. I made an example in the example app that shows this quick and dirty sidebar demo.

![sidebar demo](https://user-images.githubusercontent.com/9063669/212788021-f3dfa840-25aa-49fe-aa8e-ab3f4e4a046b.gif)

Reply:
Good point, i think we should reset height on transitionEnd then I'll check other scenarios

Reply:
Actually responsive and fixed value of height will work perfectly without saving initial height.
The only issue that we will have is with the auto height, it is changing while animating width, so i think in this case it is up to the user  to set the option **collapseStyles** what do you think ?

![colla](https://user-images.githubusercontent.com/20561190/213032923-6d82b946-6130-4ef7-8630-26a60fcc51a0.gif)